rrfs-test/validated_yamls/templates/obtype_config/msonet_stationPressure_188.yaml

```diff
@@ -86,56 +102,99 @@
              is_in: 188
            action:
              name: inflate error
-             #inflation factor: 1.7143
              inflation variable:
                name: ObsFunction/ObsErrorFactorSfcPressure
-           defer to post: true
+               options:
+                 geovar_geomz: geopotential_height
+               #  geovar_sfc_geomz: surface_geometric_height
+                 station_altitude: height
 
+         # Bounds Check
          - filter: Bounds Check
            apply at iterations: 0,1
            filter variables:
            - name: stationPressure
-           minvalue: 0.0
+           minvalue: 20000.0
            maxvalue: 120000.0
 
+         # Create temporary ObsErrorData
+         - filter: Variable Assignment
+           assignments:
+           - name: TempObsErrorData/stationPressure
+             type: float
+             function:
+               name: ObsFunction/Arithmetic
+               options:
+                 variables:
+                 - name: ObsErrorData/stationPressure
+
+         # Set ObsError set "error parameter" if < "max value"
+         - filter: Perform Action
+           filter variables:
+           - name: stationPressure
+           action:
+             name: assign error
+             error parameter: &error_min 100.0
+           where:
+           - variable:
+               name: ObsErrorData/stationPressure
+             maxvalue: *error_min
+           - variable:
+               name: ObsErrorData/stationPressure
+             value: is_valid
+
+         # Set ObsError set "error parameter" if > "min value"
+         - filter: Perform Action
+           filter variables:
+           - name: stationPressure
+           action:
+             name: assign error
+             error parameter: &error_max 300.0
+           where:
+           - variable:
+               name: ObsErrorData/stationPressure
+             minvalue: *error_max
+           - variable:
+               name: ObsErrorData/stationPressure
+             value: is_valid
+
+         # Gross Error Check
          - filter: Background Check
            apply at iterations: 0,1
            filter variables:
            - name: stationPressure
-           absolute threshold: 5.0
-           threshold: 5.0
-           where:
-           - variable: ObsType/stationPressure
-             is_in: 188
+           threshold: 50.0
```

Comment:
I think the threshold should be 5 here, and 3.5 below?

Reply:
@spanNOAA, good question. I actually really wasn't too sure about this. Let me just show some results to start some discussion. I think we need to look into it some more. Neither result looks particularly great, but the magnitudes of the increments looked a little closer at x10.

**Result with threshold at x1 (5 & 3.5):** 
![fv3jedi_vs_gsi_increment_airTemperature_msonet_stationPressure_188](https://github.com/user-attachments/assets/07eeffda-2c97-4715-bbd3-7d555b273fbb)

**Result with threshold at x10 (50 & 35):** 
![fv3jedi_vs_gsi_increment_airTemperature_msonet_stationPressure_188](https://github.com/user-attachments/assets/6a4189b5-9364-4260-bb94-2950b103fc4b)

Reply:
Hmm, this is interesting. What’s behind this difference? Are there more observations passing the gross check and entering the minimization process? Have you looked into the observations assimilated by both systems in this case?

Reply:
The only difference is the cgross thresholds for the gross error checks. Increasing the cgross threshold values would be less restrictive and allow more obs through the gross check. I haven't looked at the observations themselves, but I have ob counts (nobs) in each figure.

Reply:
@spanNOAA, I'm adding print statements in GSI:setupps.f90 and UFO:BackgroundCheck.cc to look into this more with a single ob test. 

There are these lines in setupps.f90 which draw my attention
```
obserror = min(r10/max(ratio_errors*error,tiny_r_kind),huge_r_kind)
residual = abs(r10*ddiff)
```

My print in GSI:
```
281:    obserror = 0.541315926311159     
281:    obserrlm = 1.00000000000000     
281:    residual = 3.47478724557078     
281:    ratio = 3.47478724557078     
281:    ddiff = -0.347478724557078     
281:    error = 18.5563178156275     
```

My prints in UFO:
```
 0:   errorMultiplier[jobs] = 54.1881
 0:   hofx[jobs] - obs[jv][jobs] = 334.414
```

In GSI ratio > cgross is compared. In JEDI hofx[jobs] - obs[jv][jobs] > thr*errorMultiplier[jobs] is compared. where I have thr (threshold) set to x10 GSI's value of 5. Here is how I see the calculation going in each system:

In GSI:
```f90
obserror=0.5413
obserrlm=1
residual=10*0.3474=3.474
ratio=3.474/1

#Then check if ratio > cgross
3.474 > 5
```

In UFO:
```f90
thr=50 #I got 50 by taking cgross in convinfo and x10
errorMultiplier=54.18
hofx - obs = 334.414

#Then check if hofx - obs - bias) > thr * errorMultiplier
334.414/54.18 > 50
```

I'm thinking a couple of things. The thing I am leaning closer toward is that the errorMultiplier in UFO just needs to use the 100 Pa min error that I was trying to set in the yaml. That would then make the calculation such that I can compare against 5 instead of 50.

Reply:
It seems that errorMultiplier depends on thresholdWrtBGerror to determine whether to use hofxerr or obserr_[iv]. What I’m wondering is: what exactly is thresholdWrtBGerror, and how is it configured? In GSI, the ratio is calculated as abs(pob - pges) in hPa divided by obserrlm, which is set to errmin since obserr is smaller than errmin (which is 1). It appears that errorMultiplier is intended to function similarly to obserrlm in GSI, but its value seems to be equal to obserr in JEDI.

Reply:
hofxerr is only used if you specify it in the yaml. We are not. You have to specify it as:
```
threshold wrt background error: true
```
So you can use the background error instead of the obserror.

I'm thinking that the min ob error of 100 Pa isn't getting into the Background Check. When I remove the "Create temporary ObsErrorData" part, it does use the 100 Pa. The only thing I can think of is the Background Check tries using that TempObsErrorData/stationPressure values instead of ObsErrorData/stationPressure. My understanding is that it should use ObsErrorData unless otherwise specified.

Reply:
@spanNOAA, after trying a few things this was the best I could get it to be using the same thresholds as GSI. There could very well be something I'm missing too. The ob counts actually look a lot closer, but the spatial structure and magnitude of the increments still looks a little off. Perhaps it is slightly closer to GSI now than in the other two options I have shared in this thread? Other than now having the thresholds the same as GSI, I just added `defer to post: true` after each step starting with creating the TempObsErrorData, setting max/min oberrors for gross checks, the gross checks themselves, and re-assigning the TempObsErrorData back into ObsErrorData. I also put some debug print statements in the code to make sure that the min/max error values to be used in the gross check is being used. I was finding that without the `defer to post: true` it was just using the TempObsErrorData values in the gross error check--hence the reason I needed to have a larger cgross value.

 
![fv3jedi_vs_gsi_increment_airTemperature_msonet_stationPressure_188](https://github.com/user-attachments/assets/32987b0e-260f-4b73-91fe-8a3f6bed5d3c)